src/v2/security/PermissionService.ts

```diff
@@ -1,16 +1,16 @@
 import { Injectable } from '@nestjs/common';
-import { StudentRepository } from '../api/user/StudentRepository';
 import { GrantData } from '../api/user/role/data/GrantData';
 import { RoleData } from '../api/user/role/data/RoleData';
+import { UserService } from '../api/user/UserService';
 
 @Injectable()
 export class PermissionService {
   constructor (
-    private studentRepository: StudentRepository,
+    private userService: UserService,
   ) {}
 
   async hasPermission (userId: string, permission: string) {
-    const roles = await this.studentRepository.getRoles(userId);
+    const roles = await this.userService.getRoles(userId);
```

Comment:
you can use roleRepository for such purposes. it's better solution instead of service